base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1082,28 +1084,47 @@ pub unsafe extern "C" fn seed_words_push_word(
         },
     }
 
-    if MnemonicLanguage::from(word_string.as_str()).is_err() {
-        log::error!(target: LOG_TARGET, "{} is not a valid mnemonic seed word", word_string);
-        return SeedWordPushResult::InvalidSeedWord as u8;
+    // Seed words is currently empty, this is the first word
+    if (*seed_words).0.is_empty() {
+        (*seed_words).0.push(word_string);
+        return SeedWordPushResult::SuccessfulPush as u8;
     }
 
-    (*seed_words).0.push(word_string);
-    if (*seed_words).0.len() >= 24 {
-        return if let Err(e) = CipherSeed::from_mnemonic(&(*seed_words).0, None) {
-            log::error!(
-                target: LOG_TARGET,
-                "Problem building valid private seed from seed phrase: {:?}",
-                e
-            );
-            error = LibWalletError::from(WalletError::KeyManagerError(e)).code;
-            ptr::swap(error_out, &mut error as *mut c_int);
-            SeedWordPushResult::InvalidSeedPhrase as u8
+    // Try push to a temporary copy first to prevent existing object becoming invalid
+    let mut temp = (*seed_words).0.clone();
+    temp.push(word_string.clone());
+
+    // Check words in temp are still consistent for a language
+    if let Ok(_language) = MnemonicLanguage::detect_language(&temp) {
+        if temp.len() >= 24 {
+            if let Err(e) = CipherSeed::from_mnemonic(&temp, None) {
+                log::error!(
+                    target: LOG_TARGET,
+                    "Problem building valid private seed from seed phrase: {:?}",
+                    e
+                );
+                error = LibWalletError::from(WalletError::KeyManagerError(e)).code;
+                ptr::swap(error_out, &mut error as *mut c_int);
+                return SeedWordPushResult::InvalidSeedPhrase as u8;
+            };
+        }
+
+        (*seed_words).0.push(word_string);
+
+        // Note: test for a validity was already done so we can just check length here
+        if (*seed_words).0.len() < 24 {
+            SeedWordPushResult::SuccessfulPush as u8
         } else {
             SeedWordPushResult::SeedPhraseComplete as u8
-        };
+        }
+    } else {
+        log::error!(
+            target: LOG_TARGET,
+            "Word in seed phrase is inconsistent: {:?}",
```

Comment:
Do you mean that there is no language matching the words so far? If so I think this error is not very clear. If you saw it in the logs in isolation would you be able to determine what the problem is? Maybe something like "Current words in seed phrase do not match a supported language"

Reply:
What is meant here is that by adding the word, the language cannot be determined anymore since there are no words that are either in one language or words common to at least one of them. The error will be UnknownLanguage.

Perhaps a pre-detection of the language prior to this and adding "...previously detect words were in `language`" will help?